### PR TITLE
Print keys on log as hex string

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve? 
Done the backlog (6) in #259 

### What is changed and how it works?
Replace all of these:
`zap.(Binary|ByteString)\("(.*)", (.*)\)` => `zap.Stringer("$1", utils.WrapKey($2))`
`zap.ByteStrings\("(.*)", (.*)\)` => `zap.Array("$1", utils.WrapKeys($2))`
`zap.Any("sstMeta", (.*))` => `utils.ZapSSTMeta($1)`
`zap.Any("region", (.*))` => `utils.ZapRegion($1)`
`zap.Any("rewriteRule", (.*))` => `utils.ZapRewriteRule($1)`
`zap.Any("file", (.*))` => `utils.ZapFile($1)`

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
    - See log files provided below.

### Release Note

 - Unify the form of keys in logger to hex string. 

### Note

Here are some simple logs after changed, logs with '_r' suffix are restore logs:
[log_first.log](https://github.com/pingcap/br/files/4784565/log_first.log)
[log_first_r.log](https://github.com/pingcap/br/files/4784566/log_first_r.log)
[log_second.log](https://github.com/pingcap/br/files/4784567/log_second.log)
[log_second_r.log](https://github.com/pingcap/br/files/4784568/log_second_r.log)

### CODE REVIEW

**Comment by YuJuncen** [comment]
Even @kennytm suggested that we should use upper-case hex string, due to `kv.Key.String` calls `hex.EncodeToString` directly, which produces a lower-case hex string.

Should we write another `Stringer`?

File: pkg/utils/logging.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+// WrapKeys wrap keys as a ArrayMarshaler that can print proper upper hex format.
+func WrapKeys(keys [][]byte) zapcore.ArrayMarshaler {
+	return zapArrayMarshalKeysMixIn(keys)
+}
+
+// ZapRewriteRule make the zap fields for a rewrite rule.
+func ZapRewriteRule(rewriteRule *import_sstpb.RewriteRule) zapcore.Field {
+	return zap.Object("rewrite rule", zapMarshalRewriteRuleMixIn{rewriteRule})
```
Comment (kennytm) [inline pkg/utils/logging.go:100]:
```suggestion
	return zap.Object("rule", zapMarshalRewriteRuleMixIn{rewriteRule})
```

File: pkg/utils/logging.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+// ZapFile make the zap fields for a file.
+func ZapFile(file *backup.File) zapcore.Field {
+	return zap.Object("file", zapMarshalFileMixIn{file})
+}
+
+// ZapSSTMeta make the zap fields for a SST meta.
+func ZapSSTMeta(sstMeta *import_sstpb.SSTMeta) zapcore.Field {
+	return zap.Object("sst meta", zapMarshalSSTMetaMixIn{sstMeta})
```
Comment (kennytm) [inline pkg/utils/logging.go:115]:
```suggestion
	return zap.Object("sstMeta", zapMarshalSSTMetaMixIn{sstMeta})
```

File: pkg/utils/logging.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	enc.AddString("name", file.GetName())
+	enc.AddString("CF", file.GetCf())
+	enc.AddString("sha256", hex.EncodeToString(file.GetSha256()))
+	enc.AddString("start key", WrapKey(file.GetStartKey()).String())
+	enc.AddString("end key", WrapKey(file.GetEndKey()).String())
+	enc.AddUint64("start version", file.GetStartVersion())
+	enc.AddUint64("end version", file.GetEndVersion())
+	enc.AddUint64("total kvs", file.GetTotalKvs())
+	enc.AddUint64("total bytes", file.GetTotalBytes())
+	enc.AddUint64("CRC64 xor", file.GetCrc64Xor())
```
Comment (kennytm) [inline pkg/utils/logging.go:31]: please make the labels the same as before. i highly doubt the old output contained spaces like `"start key"`.

**Comment by kennytm** [comment]
Failed to download TiFlash.zip 🤔

**Pull request merged**